src/Tools/CodeGenerator/Featurizers.yaml

```diff
@@ -57,10 +57,88 @@ featurizers:
       vocabulary established during training.
 
       C++-style pseudo signature:
-        TODO
+        CountVector execute(std::string const &value);
 
       Examples:
-        TODO
+        Assuming the training data is...
+        ["orange apple orange grape", "grape carrot carrot apple", "peach banana orange banana"]
+        The input data is...
+        ["banana grape grape apple apple apple orange"]
+        The result will be computed by...
+          categorize and compute each word's number of apperance in input data, we have "apple -> 3", "banana -> 1", "grape -> 2", "orange -> 1"
+          construct a dictionary and assign id for each unique word using training data, we have "apple -> 0", "banana -> 1", "grape -> 3", "orange -> 4"
+          generate TFStruct by combining <word's id, word's number of apperance>
+        The result is...
+        [TFStruct(0, 3), TFStruct(3, 2), TFStruct(1, 1), TFStruct(4, 1)]
+
+    type_mappings:
+      - input_type: std::string
+        output_type: CountVectorEncoding
+
+    configuration_params:
+      - type: bool
+        name: lowercase
+        is_optional: false
+
+      - type: int
+        name: analyzer
+        is_optional: false
+
+      - type: string
+        name: regexToken
+        is_optional: false
+
+      - type: IndexMap
+        name: existingVocabulary
+        is_optional: true
+
+      - type: uint32
+        name: topKTerms
+        is_optional: true
+
+      - type: float
+        name: minDf
+        is_optional: false
+
+      - type: float
+        name: maxDf
+        is_optional: false
+
+      - type: uint32
+        name: ngramRangeMin
+        is_optional: false
+
+      - type: uint32
+        name: ngramRangeMax
+        is_optional: false
+
+      - type: bool
+        name: binary
+        is_optional: false
+
+
+    custom_structs:
+      - name: CountVectorEncoding
+        description: |-
+          Sparse vector of a matrix of token counts
+
+        members:
+          - type: uint32
+            name: NumElements
+          - type: uint32
+            name: Value
+          - type: uint32
+            name: Index
+
+      - name: IndexMap
+        description: |-
+          map<string, uint32>
+
+        members:

```

Comment:
This is going to return a single value. We will need to talk about this return type.